From 2634cb60f8c0183e87710ab53b43d1956dc61b33 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:36:09 -0400
Subject: [ethernet] i40e: reset VFs after PF reset

Message-id: <1409658042-22689-9-git-send-email-sassmann@redhat.com>
Patchwork-id: 90202
O-Subject: [RHEL7.1 PATCH 008/281] i40e: reset VFs after PF reset
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

Reset all of the VFs after a PF reset, so that they are in a known
state, and the VF driver can detect the reset and reinit itself.

Change-ID: I93c5b3a0f8b1371d0da078f92de948b9d3a6413f
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 69f64b2b77012e5e1d9ae25135e85aa7f469c397)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_main.c b/drivers/net/ethernet/intel/i40e/i40e_main.c
index 8e44411..21d46f4 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_main.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_main.c
@@ -5332,6 +5332,11 @@ static void i40e_reset_and_rebuild(struct i40e_pf *pf, bool reinit)
  /* restart the VSIs that were rebuilt and running before the reset */
  i40e_pf_unquiesce_all_vsi(pf);
 
+ if (pf->num_alloc_vfs) {
+  for (v = 0; v < pf->num_alloc_vfs; v++)
+   i40e_reset_vf(&pf->vf[v], true);
+ }
+
  /* tell the firmware that we're starting */
  dv.major_version = DRV_VERSION_MAJOR;
  dv.minor_version = DRV_VERSION_MINOR;
-- 
1.7.1